hoot-core/src/main/cpp/hoot/core/io/OsmXmlWriter.h

```diff
@@ -112,16 +113,28 @@ class OsmXmlWriter : public QXmlDefaultHandler, public PartialOsmMapWriter
   virtual void writePartial(const ConstRelationPtr& relation);
   virtual void finalizePartial();
 
-  /**
-   * Remove any invalid characters from the string s and print an error if one is found.
-   */
-  QString removeInvalidCharacters(const QString& s);
-
   bool getFormatXml() const { return _formatXml; }
   void setFormatXml(const bool format) { _formatXml = format; }
 
   virtual QString supportedFormats() { return ".osm"; }
 
```

Comment:
Some chars should never be in xml and should be removed (would only be there if picked up from a non-xml data source), so use removeInvalidCharacters for taking care of that.  Other chars just need to be encoded in data to make the xml well-formed, so use encodeData for that.  The reason encoding the data is optional is b/c in certain situations, JSON is being embedded in XML output (presumably by a translation script somewhere?...couldn't find where it was happening).  In that case, you don't want to encode things like quotes which are needed by JSON.